x-pack/plugins/transform/server/routes/api/transforms.ts

```diff
@@ -569,14 +558,16 @@ const startTransformsHandler: RequestHandler<
 
 async function startTransforms(
   transformsInfo: StartTransformsRequestSchema,
-  callAsCurrentUser: LegacyAPICaller
+  esClient: ElasticsearchClient
```

Comment:
if you adopt the change suggested above, only `transformClient` could be passed in to these functions.